tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -1526,23 +1538,65 @@ pub(super) mod test {
             .with_timer(MockTime);
         let expected = "fake time tracing_subscriber::fmt::format::test: hello\n";
 
-        run_test(subscriber, make_writer, expected);
+        assert_info_hello(subscriber, make_writer, expected);
     }
 
-    #[cfg(feature = "ansi")]
-    fn test_ansi(is_ansi: bool, expected: &str) {
+    #[test]
+    fn with_thread_ids() {
         let make_writer = MockMakeWriter::default();
         let subscriber = crate::fmt::Collector::builder()
             .with_writer(make_writer.clone())
-            .with_ansi(is_ansi)
+            .with_thread_ids(true)
+            .with_ansi(false)
+            .with_timer(MockTime);
+        let expected =
+            "fake time  INFO ThreadId(NUMERIC) tracing_subscriber::fmt::format::test: hello\n";
+
+        assert_info_hello_ignore_numeric(subscriber, make_writer, expected);
+    }
+
+    #[test]
+    fn pretty_default() {
+        let make_writer = MockMakeWriter::default();
+        let subscriber = crate::fmt::Collector::builder()
+            .pretty()
+            .with_writer(make_writer.clone())
+            .with_ansi(false)
             .with_timer(MockTime);
-        run_test(subscriber, make_writer, expected)
+        let expected = format!(
+            r#"  fake time  INFO tracing_subscriber::fmt::format::test: hello
+    at {}:NUMERIC
+
+"#,
+            file!()
+        );
+
+        assert_info_hello_ignore_numeric(subscriber, make_writer, &expected)
+    }
+
+    fn assert_info_hello(subscriber: impl Into<Dispatch>, buf: MockMakeWriter, expected: &str) {
+        let _default = set_default(&subscriber.into());
+        tracing::info!("hello");
+        let result = buf.get_string();
+
+        assert_eq!(expected, result)
     }
 
-    fn run_test(subscriber: impl Into<Dispatch>, buf: MockMakeWriter, expected: &str) {
+    // When numeric characters are used they often form a non-deterministic value as they usually represent things like a thread id or line number.
+    // This assert method should be used when non-deterministic numeric characters are present.
+    fn assert_info_hello_ignore_numeric(
+        subscriber: impl Into<Dispatch>,
+        buf: MockMakeWriter,
+        expected: &str,
+    ) {
         let _default = set_default(&subscriber.into());
         tracing::info!("hello");
-        assert_eq!(expected, buf.get_string())
+
+        let regex = Regex::new("[0-9]+").unwrap();
+        let result = buf.get_string();
+        let result_cleaned = regex.replace_all(&result, "NUMERIC");
```

Comment:
nit, take it or leave it: we _may_ want to put the regex in a `lazy_static!` to avoid recompiling the regex in every test:
```suggestion
        lazy_static::lazy_static! {
            static ref REGEX: Regex = Regex::new("[0-9]+").unwrap();
        }
            
        let result = buf.get_string();
        let result_cleaned = regex.replace_all(&REGEX, "NUMERIC");
```

in theory, this might improve performance somewhat. in practice, I don't think the regex is compiled enough to have a significant impact on how long it takes to run these tests, so this may not really be worth the effort.